### DISCUSSION TITLE
'Program run without the Pulumi engine available` error using Automation API?

### DISCUSSION THREAD

**Discussion by CyrineG** [pulumi/pulumi#9021] [Automation API] [answered]
Hello! I'm writing a Flask app (inside a docker container) to provision scaleway instances. I'm using Pulumi Automation API, and I keep getting a `Program run without the Pulumi engine available; re-run using the pulumi CLI` error. The pulumi CLI is available inside my container and I'm following the [Automation API project examples](https://github.com/komalali/self-service-platyform). 

root@4aab168725c7:/code-provisioning# pulumi version
v3.24.1

I'm new to pulumi so I might be doing things very wrong, but I thought the Automation API uses the CLI on its own?

**Comment by cnunciato** [comment, marked as answer]
Hi @CyrineG! Any chance you could share some of your program code? And maybe a portion of the Dockerfile you're working with? The code in the repo you linked to works for me (with a minor adjustment to fix what looks like a [recent change in one of the program's transitive dependencies](https://github.com/aws/aws-sam-cli/issues/3661)):

```bash
$ python3 -m venv venv
$ venv/bin/python3 -m pip install --upgrade pip
$ venv/bin/pip install -r requirements.txt
$ venv/bin/pip install markupsafe==2.0.1
$ source venv/bin/activate
...

$ FLASK_RUN_PORT=1337 FLASK_ENV=development FLASK_APP=app PULUMI_ORG=cnunciato flask run
 * Serving Flask app "app" (lazy loading)
 * Environment: development
 * Debug mode: on
 * Running on http://127.0.0.1:1337/ (Press CTRL+C to quit)
...
```

... so I suspect it may have something to do with your Docker configuration.

**Reply by CyrineG** [reply]
Hello, thank you for your reply! I found the issue, it was because I was passing a parameter to the program function in  create_stack
`
        stack = automation.create_stack(stack_name=stack_name,
                                  project_name=project_name,
                                  program=create_pulumi_program(content))`

It should be instead like this: 
`
        stack = automation.create_stack(stack_name=stack_name,
                                  project_name=project_name,
                                  program=create_pulumi_program)`

**Comment by nimbinatus** [comment]
(note that I can't select comments in a thread as an answer, so please check the thread!)
